src/views/Dash.tsx

```diff
@@ -38,13 +39,34 @@ export default function Dash() {
     setProjectDetails(details);
   }
 
+  function validityCheck(isValid: boolean) {
```

Comment:
Make this an async function...remove the many `.then`'s and use `await` instead.